torch/fx/experimental/migrate_gradual_types/constraint_generator.py

```diff
@@ -272,15 +274,50 @@ def gt_inference_rule(n: Node, symbols, constraints, counter):
 def lt_inference_rule(n: Node, symbols, constraints, counter):
     raise NotImplementedError('Not yet implemented')
 
-# TODO
+
 @register_inference_rule(torch.full)
 def full_inference_rule(n: Node, symbols, constraints, counter):
-    raise NotImplementedError('Not yet implemented')
+    full, counter = gen_tvar(counter)
+    symbols[n] = full
+    res = []
+    assert isinstance(n.args[0], Iterable)
+    for arg in n.args[0]:
+        res.append(symbols[arg])
+    c = BinConstraintT(full, TensorType(list(res)), op_eq)  # type: ignore[arg-type]
+    return [c], counter
 
-# TODO
+
+# TODO normalize index
 @register_inference_rule(torch.arange)
 def arange_inference_rule(n: Node, symbols, constraints, counter):
-    raise NotImplementedError('Not yet implemented')
+    start = 0
+    step = 1
+
+    if len(n.args) == 1:
+        end = symbols[n.args[0]]
+    else:
+        raise NotImplementedError('Not yet implemented')
+
+    # int((end - start) / step)
+    d1, counter = gen_dvar(counter)
+    size_constraint = BinConstraintD(d1, BinConstraintD(BinConstraintD(end, start, op_sub), step, op_div), op_eq)
+    arange, counter = gen_tvar(counter)
+    symbols[n] = arange
+
+    # either the a parameter is a number or it is Dyn
+    c1 = Disj([BinConstraintD(end, Dyn, op_eq),
+               BinConstraintD(start, Dyn, op_eq),
+               BinConstraintD(step, Dyn, op_eq)])
+    c2 = BinConstraintD(d1, Dyn, op_eq)
+    both_dyn = Conj([c1, c2])
+
+    c11 = Conj([BinConstraintD(end, Dyn, op_neq),
+                BinConstraintD(start, Dyn, op_neq),
+                BinConstraintD(step, Dyn, op_neq)])
+    c22 = BinConstraintD(d1, Dyn, op_neq)
+    both_numbers = Conj([c11, c22, size_constraint])
```

Comment:
IMO this would all be easier to read if we had operator overloads, e.g.

```
size_constraint = d1 == ((end - start) / step)
c1 = (end == Dyn) || (start == Dyn) || (step == Dyn)
c2 = (d1 == Dyn)
both_dyn = c1 && c2

c11 = (end != Dyn) && (start != Dyn) && (step != Dyn)
c22 = (d1 != Dyn)
both_numbers = c11 && c22 && 
```

Is it possible to implement those?

Reply:
We discussed (still figuring it out) and I will revisit it in a new PR